### PR TITLE
fix: Transform order in subassembly

### PR DESCRIPTION
**Pull request by lenianiva**
Fixes https://github.com/CadQuery/cadquery/issues/1628

Ideally I want to add some unit tests but I don't know the best way to approach this. Maybe test for the intersection volume of the black and white arrows?

### CODE REVIEW

**Comment by adam-urbanczyk** [comment]
Thanks @lenianiva ! As you mentioned in #1628 you could add a test asserting if the sublocs are equal.

**Comment by lenianiva** [comment]
> Thanks @lenianiva ! As you mentioned in #1628 you could add a test asserting if the sublocs are equal.

I think testing exact equality doesn't work. I added a really simple example which has different behaviours under the old and new implementations, and the coordinates don't fully match up.
```python
import cadquery as cq
from functools import reduce
from typing import Tuple, cast

def f():
    cube = cq.Solid.makeBox(1, 1, 1)
    inner = (
        cq.Assembly()
        .add(cube, name="c1")
        .add(cube, name="c2", loc=cq.Location((0, 0, 1), (0, 1, 0), 45))
    )
    middle = (
        cq.Assembly()
        .add(inner, name="inner", loc=cq.Location((0, 1, 0), (0, 0, 1), 30))
    )
    outer = (
        cq.Assembly()
        .add(middle, name="middle", loc=cq.Location((0,0,0)))
    )


    #cq.Assembly._subloc = _subloc
    name, _ = middle._query("inner/c2")
    native_loc = middle.objects[name].loc
    print(native_loc.toTuple())
    loc, _ = middle._subloc(name)
    print(loc.toTuple())


    name, _ = outer._query("middle/inner/c2")
    native_loc = outer.objects[name].loc
    print(native_loc.toTuple())
    loc, _ = outer._subloc(name)
    print(loc.toTuple())

f()
```

**Comment by lenianiva** [comment]
@adam-urbanczyk I don't think using approx is the issue here, but the location from `_query` and `_subloc` each encode a part of the relative position of the referenced anchor. e.g. the following generates the absolute location of an object
```python
    result = ... # some assembly
    key = "assembly/obj?tag"
    name, sh = result._query(key)
    loc_self = sh.location()
    loc_parent, _ = result._subloc(name)
    loc = loc_parent * loc_self
    print(name, loc.toTuple())
```

**Comment by adam-urbanczyk** [comment]
Yup, trying to fix the test. I distilled the example made by @lorenzncode :
afbeelding
Those two spheres should coincide after the fix, and this is exactly what the test asserts now.

**Review by adam-urbanczyk — APPROVED** [review]

File: tests/test_assembly.py [hunk 1]
```diff
@@ -1610,29 +1610,33 @@ def test_imprinting(touching_assy, disjoint_assy):
...
-    name, _ = middle._query("inner/c2")
-    loc1, _ = middle._subloc(name)
+    # attach the second marker to the tagged corner, but this time with nesting
+    assy2.constrain("assy1/marker1", "Fixed")
+    assy2.constrain("marker2", "assy1/assy0/part1?vtag", "Point")
+    assy2.solve()
 
-    name, _ = outer._query("middle/inner/c2")
-    loc2, _ = outer._subloc(name)
+    # marker1 and marer2 should coincide
```
Comment (lorenzncode) [inline tests/test_assembly.py:1641]: typo in comment *marker2*

**Review by lorenzncode — APPROVED** [review]
LGTM, just one typo in test comment.

File: tests/test_assembly.py (hunk 1 above)
Comment (adam-urbanczyk) [reply tests/test_assembly.py]: Thanks, solved

**Comment by adam-urbanczyk** [comment]
Thanks @lenianiva and @lorenzncode !

**Pull request merged**